File: app/src/main/java/com/example/weizhenbin/wangebug/modules/code/uis/CodeSystemFragment.java
package com.example.weizhenbin.wangebug.modules.code.uis;

import android.graphics.drawable.Drawable;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.example.weizhenbin.wangebug.R;
import com.example.weizhenbin.wangebug.base.BaseFragment;
import com.example.weizhenbin.wangebug.base.DataResultAdapter;
import com.example.weizhenbin.wangebug.modules.code.adapters.CodeSystemTreeAdapter;
import com.example.weizhenbin.wangebug.modules.code.controllers.CodeController;
import com.example.weizhenbin.wangebug.modules.code.entity.SystemTreeDataBean;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by weizhenbin on 2018/8/24.
 */

public class CodeSystemFragment extends BaseFragment{


    SwipeRefreshLayout srlRefresh;
    RecyclerView rvDataList;
    CodeSystemTreeAdapter listAdapter;
    List<SystemTreeDataBean.DataBean> datasBeen=new ArrayList<>();




    @Override
    protected int getContentViewLayoutId() {
        return R.layout.fm_code_system;
    }

    @Override
    protected void initFragment(View view) {
        initViews(view);
        initData();
        initEvent();
    }

    private void getData() {
        CodeController.getSystemTreeData(new DataResultAdapter<SystemTreeDataBean>(){
            @Override
            public void onStart() {
                super.onStart();
                    srlRefresh.setRefreshing(true);
            }

            @Override
            public void onSuccess(SystemTreeDataBean systemTreeDataBean) {
                super.onSuccess(systemTreeDataBean);
                srlRefresh.setRefreshing(false);
                if (systemTreeDataBean!=null&&systemTreeDataBean.getErrorCode()==0){
                    if (systemTreeDataBean.getData()!=null){
                        datasBeen.clear();
                        datasBeen.addAll(systemTreeDataBean.getData());
                        listAdapter.notifyDataSetChanged();
                    }
                }
            }
        });
    }

    private void initEvent() {
        srlRefresh.setOnRefreshListener(new SwipeRefreshLayout.OnRefreshListener() {
            @Override
            public void onRefresh() {
                getData();
            }
        });
    }

    private void initData() {
        if (getContext()==null){
            return;
        }
        listAdapter=new CodeSystemTreeAdapter(datasBeen);
        rvDataList.setAdapter(listAdapter);
        rvDataList.setLayoutManager(new LinearLayoutManager(getContext(),LinearLayoutManager.VERTICAL,false));
        DividerItemDecoration itemDecoration=new DividerItemDecoration(getContext(),DividerItemDecoration.VERTICAL);
        Drawable drawable=getResources().getDrawable(R.drawable.divider_line);
        itemDecoration.setDrawable(drawable);
        rvDataList.addItemDecoration(itemDecoration);
    }

    private void initViews(View view) {
        srlRefresh=view.findViewById(R.id.srl_refresh);
        rvDataList=view.findViewById(R.id.rv_data_list);
    }


    @Override
    protected void loadData() {
        if (datasBeen.isEmpty()) {
            getData();
        }
    }

    @Override
    protected String getPageTitle() {
        return "体系";
    }

    public static CodeSystemFragment getFragment(){
        return new CodeSystemFragment();
    }


}

File: app/src/main/java/com/example/weizhenbin/wangebug/modules/code/uis/CodeSystemFragment.kt
package com.example.weizhenbin.wangebug.modules.code.uis

import android.support.v4.widget.SwipeRefreshLayout
import android.support.v7.widget.DividerItemDecoration
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.view.View
import com.example.weizhenbin.wangebug.R
import com.example.weizhenbin.wangebug.base.BaseFragment
import com.example.weizhenbin.wangebug.base.DataResultAdapter
import com.example.weizhenbin.wangebug.modules.code.adapters.CodeSystemTreeAdapter
import com.example.weizhenbin.wangebug.modules.code.controllers.CodeController
import com.example.weizhenbin.wangebug.modules.code.entity.SystemTreeDataBean
import java.util.*

/**
 * Created by weizhenbin on 2018/8/24.
 */

class CodeSystemFragment : BaseFragment() {


    lateinit var srlRefresh: SwipeRefreshLayout
    lateinit var rvDataList: RecyclerView
    lateinit var listAdapter: CodeSystemTreeAdapter
    internal var dataBeen: MutableList<SystemTreeDataBean.DataBean> = ArrayList()


    override val contentViewLayoutId: Int
        get() = R.layout.fm_code_system

    override var pageTitle: String?
        get() = "体系"
        set(value) {
            super.pageTitle = value
        }

    override fun initFragment(view: View?) {
        initViews(view!!)
        initData()
        initEvent()
    }

    private fun getData() {
        CodeController.getSystemTreeData(object : DataResultAdapter<SystemTreeDataBean>() {
            override fun onSuccess(systemTreeDataBean: SystemTreeDataBean?) {
                super.onSuccess(systemTreeDataBean)
                srlRefresh.isRefreshing = false
                if (systemTreeDataBean != null && systemTreeDataBean.errorCode == 0) {
                    if (systemTreeDataBean.data?.isEmpty() != null) {
                        dataBeen.clear()
                        dataBeen.addAll(systemTreeDataBean.data!!)
                        listAdapter.notifyDataSetChanged()
                    }
                } else {
                    listAdapter.emptyData()
                }
            }

            override fun onError(throwable: Throwable) {
                super.onError(throwable)
                srlRefresh.isRefreshing = false
                listAdapter.emptyData()
            }
        })
    }

    private fun initEvent() {
        srlRefresh.setOnRefreshListener { getData() }
        listAdapter.setAction { getData() }
    }

    private fun initData() {
        if (context == null) {
            return
        }
        listAdapter = CodeSystemTreeAdapter(context, dataBeen)
        rvDataList.adapter = listAdapter
        rvDataList.layoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
        val itemDecoration = DividerItemDecoration(context!!, DividerItemDecoration.VERTICAL)
        val drawable = resources.getDrawable(R.drawable.divider_line)
        itemDecoration.setDrawable(drawable)
        rvDataList.addItemDecoration(itemDecoration)
    }

    private fun initViews(view: View) {
        srlRefresh = view.findViewById(R.id.srl_refresh)
        rvDataList = view.findViewById(R.id.rv_data_list)
    }


    override fun loadData() {
        if (dataBeen.isEmpty()) {
            getData()
        }
    }

    companion object {

        val fragment: CodeSystemFragment
            get() = CodeSystemFragment()
    }


}
